modules/common/security/networking.nix

```diff
@@ -0,0 +1,201 @@
+# Copyright 2024-2025 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  ...
+}: let
+  cfg = config.ghaf.security.sysctl.network;
+in {
+  ## Options to enable IP security
+  options.ghaf.security.sysctl.network = {
+    disable-all = lib.mkOption {
+      description = ''
+        Disable all sysctl network security configuration.
+      '';
+      type = lib.types.bool;
+      default = false;
+    };
+
+    disable-ipv6 = lib.mkOption {
+      description = ''
+        Disables Internet Protocol v6.
+        It is recommended to reduce attack surface by disabling IPv6 if it is not being used.
+        References:
+        https://www.tenable.com/audits/items/CIS_AlmaLinux_OS_8_Server_v1.0.0_L2.audit:a532993e3ac29c19c0d491463a881303
+        https://linux-audit.com/linux-security-guide-for-hardening-ipv6/
+      '';
+      type = lib.types.bool;
+      default = true;
+    };
+
+    prevent-syn-flooding = lib.mkOption {
+      description = ''
+        Prevent SYN packet flooding.
+        It is recommended to reduce attack surface by disabling IPv6 if it is not being used.
+        References:
+        https://www.tenable.com/audits/items/CIS_AlmaLinux_OS_8_Server_v1.0.0_L2.audit:a532993e3ac29c19c0d491463a881303
+        https://linux-audit.com/linux-security-guide-for-hardening-ipv6/
+      '';
+      type = lib.types.bool;
+      default = true;
+    };
+
+    enable-RFC_1337-protection = lib.mkOption {
+      description = ''
+        Enable RFC 1337 fix to protect against TIME-WAIT assassination attacks or unintended connection resets.
+        References:
+        https://wiki.archlinux.org/title/Talk:Sysctl
+        https://datatracker.ietf.org/doc/html/rfc1337
+      '';
+      type = lib.types.bool;
+      default = true;
+    };
+
+    enable-rp-filter = lib.mkOption {
```

Comment:
Doesn't linux has enabled this by default anyway?

Reply:
It is not enabled by default in Linux. Since it is recommended enable rp_filter in many Linux security audits, so I thought to include it. Now it defaults to `false`.

Reply:
Seems this flag is not used? Also, NixOS seems to set it to 2 (loose mode), whereas this option sets it to 1 (strict mode). Not entirely sure about the implications within Ghaf, especially internal network, can you elaborate if strict mode makes sense as a default or not?

Reply:
I studied around this. As far as what I understood, loose mode(default) is fine for VMs which are communicating through internal network. Strict mode should be beneficial in net-vm specially if some one tries to spoof an VM address. In strict mode these packets will be dropped while in loose mode these may be accepted as there is always a valid route for these.